src/Symfony/Component/TypeInfo/Tests/TypeResolver/StringTypeResolverTest.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\TypeInfo\Tests\TypeResolver;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\TypeInfo\Exception\InvalidArgumentException;
+use Symfony\Component\TypeInfo\Exception\UnsupportedException;
+use Symfony\Component\TypeInfo\Tests\Fixtures\AbstractDummy;
+use Symfony\Component\TypeInfo\Tests\Fixtures\Dummy;
+use Symfony\Component\TypeInfo\Tests\Fixtures\DummyWithTemplates;
+use Symfony\Component\TypeInfo\Type;
+use Symfony\Component\TypeInfo\TypeContext\TypeContext;
+use Symfony\Component\TypeInfo\TypeContext\TypeContextFactory;
+use Symfony\Component\TypeInfo\TypeResolver\StringTypeResolver;
+
+class StringTypeResolverTest extends TestCase
+{
+    private StringTypeResolver $resolver;
+
+    protected function setUp(): void
+    {
+        $this->resolver = new StringTypeResolver();
+    }
+
+    /**
+     * @dataProvider resolveDataProvider
+     */
+    public function testResolve(Type $expectedType, string $string, TypeContext $typeContext = null)
+    {
+        $this->assertEquals($expectedType, $this->resolver->resolve($string, $typeContext));
+    }
+
+    /**
+     * @return iterable<array{0: Type, 1: string, 2?: TypeContext}>
+     */
+    public function resolveDataProvider(): iterable
+    {
+        $typeContextFactory = new TypeContextFactory(new StringTypeResolver());
+
+        // callable
+        yield [Type::callable(), 'callable(string, int): mixed'];
+
+        // array
+        yield [Type::list(Type::bool()), 'bool[]'];
```

Comment:
I think `bool[]` should be resolved to iterable, because it's not guaranteed that it's a list. It doesn't event need to be an array.

https://phpstan.org/r/b93e871c-0f74-4ada-9e93-ea71e64dd764

Reply:
Totally agreed, I resolved it as an array then.